includes/scheduler/class-comment.php

```diff
@@ -59,8 +59,8 @@ public static function schedule_comment_activity( $new_status, $old_status, $com
 			$type = 'Update';
 			\update_comment_meta( $comment->comment_ID, 'activitypub_comment_modified', time(), true );
 		} elseif (
-			'trash' === $new_status ||
-			'spam' === $new_status
+			\in_array( $new_status, array( 'trash', 'delete', 'spam' ), true ) &&
+			\Activitypub\Comment::was_received( $comment->comment_parent )
```

Comment:
We can not get sure that it is only a federated comment if the comment above was received. Comments directly to the post can also be federated for example or comments to federated local comments.

Reply:
What I mean: It does not make any sense to have that check there, it has to be in the `should_comment_be_federated` function, otherwise it will break anyways.

Reply:
But you are right, the transition is set way too late, so maybe we should simply keep your function. Sorry :(

Reply:
Maybe [45d4271](https://github.com/Automattic/wordpress-activitypub/pull/2141/commits/45d427160d187303fc544ba3c94f2b68e12c369d) is the charm?